openassessment/workflow/models.py

```diff
@@ -348,7 +360,8 @@ def update_from_assessments(self, assessment_requirements):
             score = self.get_score(assessment_requirements, step_for_name)
             # If we found a score, then we're done
             if score is not None:
-                self.set_score(score)
+                if score.get("staff_id", None) is None:
```

Comment:
I'm not grok-ing this properly - do you mean I should write this as you've described?